src/extra/utils/functions.py

```diff
@@ -50,6 +50,8 @@ def fit_gaussian(ydata, xdata=None, p0=None, norm=False, **kwargs):
         p0 (list): A list of `[y0, A, μ, σ]` to match the arguments to
             [gaussian()][extra.utils.gaussian].
         norm (bool): Whether to fit a normalized or unnormalized Gaussian.
+        Asign (int): 1 for an upwards peak, -1 for downwards. 0 (default) allows
+            either, using a faster algorithm. Passing `bounds=` overrides this.
```

Comment:
'Asign' sounds almost like an English word so I'd explicitly mention that it's the amplitude sign. And what do you mean by faster algorithm? Adding bounds should make the algorithm converge faster so I'd expect setting a specific sign to improve performance.

Reply:
Oh, sorry I didn't read the second half of your original post :upside_down_face: Ok, that's really weird...

Reply:
I think the speedup with no bounds is weird mathematically but makes some sense computationally. `curve_fit` has different algorithms behind it, and the default one depends on whether you specify any bounds:

> With `method='lm'`, the algorithm uses the Levenberg-Marquardt algorithm through [leastsq](https://docs.scipy.org/doc/scipy-1.14.0/reference/generated/scipy.optimize.leastsq.html#scipy.optimize.leastsq). Note that this algorithm can only deal with unconstrained problems.